tests/blackbox_test.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/ignition/tests/register"
+	"github.com/coreos/ignition/tests/types"
+
+	// Register the tests
+	_ "github.com/coreos/ignition/tests/registry"
+)
+
+func (server *HTTPServer) Config(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(`{
+	"ignition": { "version": "2.0.0" },
+	"storage": {
+		"files": [{
+		  "filesystem": "root",
+		  "path": "/foo/bar",
+		  "contents": { "source": "data:,example%20file%0A" }
+		}]
+	}
+}`))
+}
+
+func (server *HTTPServer) Contents(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(`asdf
+fdsa`))
+}
+
+type HTTPServer struct{}
+
+func (server *HTTPServer) Start() {
+	http.HandleFunc("/contents", server.Contents)
+	http.HandleFunc("/config", server.Config)
+
+	s := &http.Server{Addr: ":8080"}
+	go s.ListenAndServe()
+}
+
+func TestMain(m *testing.M) {
+	server := &HTTPServer{}
+	server.Start()
+	os.Exit(m.Run())
+}
+
+func TestIgnitionBlackBox(t *testing.T) {
+	for _, test := range register.Tests[register.PositiveTest] {
+		t.Run(test.Name, func(t *testing.T) {
+			outer(t, test, false)
+		})
+	}
+}
+
+func TestIgnitionBlackBoxNegative(t *testing.T) {
+	for _, test := range register.Tests[register.NegativeTest] {
+		t.Run(test.Name, func(t *testing.T) {
+			outer(t, test, true)
+		})
+	}
+}
+
+func PreCleanup(t *testing.T) {
+	mountpoints, _ := exec.Command(
+		"findmnt", "-l", "-o", "target").CombinedOutput()
+	points := strings.Split(string(mountpoints), "\n")
+	for i := len(points) - 1; i >= 0; i-- {
+		tmpDir := os.Getenv("TMPDIR")
+		if tmpDir == "" {
+			tmpDir = "/tmp"
+		}
+		for _, s := range []string{fmt.Sprintf("%s/*", tmpDir), "*hd1p*", "*ignition*"} {
+			match, err := filepath.Match(s, points[i])
+			if err != nil {
+				t.Log(err)
+			}
+			if match {
+				_, _ = exec.Command("umount", points[i]).CombinedOutput()
+			}
+		}
+	}
+}
+
+func outer(t *testing.T, test types.Test, negativeTests bool) {
+	PreCleanup(t)
+	t.Log(test.Name)
+
+	tmpDirectory, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("failed to create a temp dir: %v", err)
+	}
+	os.Chmod(tmpDirectory, 0755)
+	_ = os.Setenv("TMPDIR", tmpDirectory)
```

Comment:
Setenv should really never fail, but if it does, better to fail and make someone fix the problem than to cause surprising behavior silently.